2021/day10/day10.java

```diff
@@ -0,0 +1,83 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.LinkedList;
+
+public class day10 {
+    public static ArrayList<ArrayList<Character>> readInput(String filename) {
+        ArrayList<ArrayList<Character>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Character> line = new ArrayList<>();
+                String s = sc.next();
+                for (int i = 0; i < s.length(); i++) {
+                    line.add(s.charAt(i));
+                }
+                input.add(line);
+            }
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    public static void syntaxScoring() {
+        ArrayList<ArrayList<Character>> input = readInput("data.txt");
+        ArrayList<Long> autocompleteScores = new ArrayList<>();
+        int syntaxScore = 0;
+
+        for (ArrayList<Character> line : input) {
+
+            LinkedList<Character> stack = new LinkedList<>();
+            Boolean isCorrupted = false;
+            long autoTmpScore = 0;
+
+            for (int i = 0; i < line.size(); i++) {
+                if (line.get(i) == '(' || line.get(i) == '<' ||
+                    line.get(i) == '[' || line.get(i) == '{')
+                    stack.addFirst(line.get(i));
+
+                else if (stack.peek() == line.get(i) - 2 || stack.peek() == line.get(i) - 1)
```

Comment:
The `-1` and `-2` are kind of magic - probably some ASCII magic, but not readable. I would probably add a function that would check each case separately here.

Reply:
Also, I personally prefer to have the `{}` around my conditions, especially if it's a condition in the middle of a list fo conditions. Otherwise, it's hard to say where it ends etc.

Reply:
I am going to keep my ASCII wizardry for now :D

Reply:
I cannot accept this in this state, such arcane logic requires at least a comment, e.g.

```java
// ASCII codes for opening and closing pairs differ either by 1 ('(' and ')') or 2 (the others)
```

Think of yourself reading this code in a month and trying to figure out what the hell happened

Reply:
I am going to leave it and add the comment